surefire-providers/surefire-junit-platform/src/test/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProviderTest.java

```diff
@@ -410,11 +410,19 @@ public void allDiscoveredTestsAreInvokedForNullArgument()
         InOrder inOrder = inOrder( runListener );
 
         ArgumentCaptor<SimpleReportEntry> report = ArgumentCaptor.forClass( SimpleReportEntry.class );
-        inOrder.verify( runListener )
+        inOrder.verify( runListener, times( 2 ) )
                 .testSetStarting( report.capture() );
-        Assertions.assertThat( report.getValue().getSourceName() )
+
+        SimpleReportEntry engine = report.getAllValues().get( 0 );
+        Assertions.assertThat( engine.getSourceName() )
+            .isEqualTo( "JUnit Jupiter" );
```

Comment:
Due to this we should chceck the XML reports.
This may appear in the report and that's wrong because it is an engine and not a class.
We have another issue for the Cucumber where the ReportEntry has to be updated.
But still not sure if this is needed to transfer engine name in this issue we have here.